### PR TITLE
SALTO-1898: go to service in zendesk

### PR DESCRIPTION
**Pull request by noamhammer**
_go to service support in zendesk_

---

_Additional context for reviewer_
There are few TODOs on some URLs in the config. We need to understand if we want to keep those URLs or not. @tomermevorach is on it

---
_Release Notes_: 
_Zendesk_support adapter_
* add go to service

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/zendesk-support-adapter/src/config.ts [hunk 1]
```diff
@@ -536,6 +593,13 @@ export const DEFAULT_TYPES: Record<string, configUtils.TypeDuckTypeConfig> = {
         { fieldName: 'id', fieldType: 'number' },
         { fieldName: 'domain_verification_code' },
       ]),
+      // TODO: weird look like group
```
Comment (alonstern) [inline packages/zendesk-support-adapter/src/config.ts:596]: Could you explain this todo? The page in the UI looks weird? (different than the one the user that modify this instance through the UI will  see?)

Comment (noamhammer) [reply packages/zendesk-support-adapter/src/config.ts:596]: That's what I wrote in the `Additional context for the reviewer` section. And yes, there is no bookmarkable URL here but there is a URL that directs to an edit page that is different from the one that the user know

File: packages/zendesk-support-adapter/src/filters/service_url.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { CORE_ANNOTATIONS, Element, isInstanceElement } from '@salto-io/adapter-api'
+import { elements as elementsUtils } from '@salto-io/adapter-components'
+import { API_DEFINITIONS_CONFIG } from '../config'
+import { FilterCreator } from '../filter'
+
+const filter: FilterCreator = ({ config, client }) => ({
+  onFetch: async (elements: Element[]) => {
```
[the remark this answers is not in the document]
Comment (noamhammer) [reply packages/zendesk-support-adapter/src/filters/service_url.ts:22]: Why do you say it?

File: packages/adapter-components/src/config/transformation.ts [hunk 3]
```diff
@@ -38,6 +39,11 @@ export type FieldTypeOverrideType = {
   restrictions?: RestrictionAnnotationType
 }
 
+type UrlService = {
+  url: string
+  urlParamsToFields?: UrlParams
```
[the remark this answers is not in the document]
Comment (noamhammer) [reply packages/adapter-components/src/config/transformation.ts:44]: I don't really need it, just wanted to keep it for consistency. Wasn't aware of the swagger thing. It seems like we don't need it in deploy for DuckType either.
Do you want me to keep it like that for consistency? or do you want me to change it?

File: packages/zendesk-support-adapter/src/filters/service_url.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { CORE_ANNOTATIONS, Element, isInstanceElement } from '@salto-io/adapter-api'
+import { elements as elementsUtils } from '@salto-io/adapter-components'
+import { API_DEFINITIONS_CONFIG } from '../config'
+import { FilterCreator } from '../filter'
+
+const filter: FilterCreator = ({ config, client }) => ({
```
[the remark this answers is not in the document]
Comment (alonstern) [reply packages/zendesk-support-adapter/src/filters/service_url.ts:21]: I think the following signature will work
```
const filterCreator: <TClient, TContext extends { apiDefinitions: configUtils.AdapterApiConfig }, TResult extends void | filter.FilterResult = void> (baseUrl: string) => filterUtils.FilterCreator<TClient, TContext, TResult> = baseUrl => ({ config }) => ({
...
```

and then in adapter.ts you can do

```
serviceUrlFilter<ZendeskClient, FilterContext, FilterResult>('baseUrl')
```

File: packages/adapter-components/src/filters/service_url.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  TResult extends void | filter.FilterResult = void
+>(baseUrl: string) => FilterCreator<TClient, TContext, TResult> = baseUrl => ({ config }) => ({
+  onFetch: async (elements: Element[]) => {
+    elements
+      .filter(isInstanceElement)
+      .forEach(instance => addUrlToInstance(instance, baseUrl, config))
+  },
+  onDeploy: async (changes: Change<InstanceElement>[]) => {
+    const relevantChanges = changes.filter(isInstanceChange).filter(isAdditionChange)
+    if (_.isEmpty(relevantChanges)) {
```
Comment (alonstern) [inline packages/adapter-components/src/filters/service_url.ts:47]: nit: I think this check is not necessary

**Review by alonstern — APPROVED** [review]

**Pull request merged**